var/spack/repos/builtin/packages/py-ipykernel/package.py

```diff
@@ -30,30 +31,34 @@ class PyIpykernel(PythonPackage):
     version('4.1.1',  sha256='d8c5555386d0f18f1336dea9800f9f0fe96dcecc9757c0f980e11fdfadb661ff')
     version('4.1.0',  sha256='e0e150ad55e487e49054efc9a4b0e2e17f27e1de77444b26760789077b146d86')
 
-    depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
-    depends_on('python@3.4:', when='@5.0:', type=('build', 'run'))
-    depends_on('python@3.5:', when='@5.2:', type=('build', 'run'))
     depends_on('python@3.7:', when='@6.0:', type=('build', 'run'))
-    depends_on('py-setuptools', type='build', when='@5:')
+    depends_on('python@3.5:', when='@5.2:', type=('build', 'run'))
+    depends_on('python@3.4:', when='@5.0:', type=('build', 'run'))
+    depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
+    depends_on('py-setuptools', when='@5:', type='build')
+    depends_on('py-jupyter-core', when='@5:', type='build')
```

Comment:
4.2+, at least in the latest version

Reply:
Yes, I forgot to add the version here. Sorry and thanks.
Done.